src/index.ts

```diff
@@ -422,15 +422,15 @@ function formatPromotedCommits(
       const fileHeader = `* ${filename}\n`;
       const byEnvironment = [...promotionsByTargetEnvironment.entries()]
         .sort(([a], [b]) => a.localeCompare(b))
-        .map(([environment, environmentPromotions]) => {
+        .map(([, environmentPromotions]) => {
```

Comment:
The output looks good. But what I'm seeing here is basically, we're keeping the data structures where promotions can contain multiple target environments and we're just eliding that from the description.  But the *reason* it's OK to elide it in the description is because we're now always passing precisely one argument name as `promotion-target-regexp` (not an actual regexp) whenever we're setting `update-promoted-values` to true — so let's lean into that more fully, replace the regexp argument with a simple target name argument (maybe combining with the boolean argument too?) and simplify all the internal data structures that are passing around maps keyed by env name to not need that any more.

... ok, maybe this doesn't quite work, because the dry run workflow runs `update-promoted-values` without the regex filter at all. But that workflow also doesn't need to generate a description of the promotions.  So we could simplify things so that if you're not passing a filter, then you don't run the codepath that generates a PR description...... as long as we're confident that that won't be turning off enough of the code to defeat the point of doing a dry run.  If that's not compelling, maybe we leave in the "pass around maps that often only have one key" everywhere, but I still think we should change the top-level argument from a regexp to a simple match, since that's the only way to feel confident that (when actually making PRs) this removal isn't eliding information.

Reply:
Yes, that makes sense.  I'll change `promotion-target-regexp` to `promotion-target` and see what else I can simplify easily